engine/utils/hardware/os_info.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <json/json.h>
+#include <string>
+#include "hwinfo/hwinfo.h"
+
+namespace hardware {
```

Comment:
I think we better move those structs into `common` folder.

Reply:
Updated